r/R/type.R

```diff
@@ -130,14 +174,64 @@ TimeType <- R6Class("TimeType",
     unit = function() TimeType__unit(self)
   )
 )
-Time32 <- R6Class("Time32", inherit = TimeType)
-Time64 <- R6Class("Time64", inherit = TimeType)
+Time32 <- R6Class("Time32",
+  inherit = TimeType,
+  public = list(
+    code = function() {
+      unit <- self$unit()
+      unit <- if (unit == TimeUnit$MILLI) {
+        "ms"
+      } else if (unit == TimeUnit$SECOND) {
+        "s"
+      } else {
+        abort(c(
+          "Invalid unit.",
+          i = 'The `Time32` type only supports "ms" (TimeUnit$MILLI) and "s" (TimeUnit$SECOND).'
+        ), call = call("code"))
+      }
+      call2("time32", unit = unit)
+    }
+  )
+)
+Time64 <- R6Class("Time64",
+  inherit = TimeType,
+  public = list(
+    code = function() {
+      unit <- self$unit()
+      unit <- if (unit == TimeUnit$NANO) {
+        "ns"
+      } else if (unit == TimeUnit$MICRO) {
+        "us"
+      } else {
+        abort(c(
+          "Invalid unit.",
```

Comment:
Do we need to validate here? If the object exists, doesn't that mean it has already been validated by Arrow?